File: 0036-valid-sudoku/0036-valid-sudoku.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        //cheacking for row
        for(int i=0;i<9;i++){
            unordered_map<char,int> count;
            for(int j=0;j<9;j++){
                char ch=board[i][j];
                if(ch!='.'){
                    if(count[ch]!=0){
                        return false;
                    }
                    count[ch]=1;
                }
            }
        }
        //for column
        for(int i=0;i<9;i++){
            unordered_map<char,int> count;
            for(int j=0;j<9;j++){
                char ch=board[j][i];
                if(ch!='.'){
                    if(count[ch]!=0){
                        return false;
                    }
                    count[ch]=1;
                }
            }
        }
        //checking 
        for(int l=0;l<3;l++){
            for(int k=0;k<3;k++){
                unordered_map<char,int> count;
                for(int i=0+3*l;i<3+3*l;i++){
                    for(int j=0+k*3;j<3+3*k;j++){
                        char ch=board[i][j];
                        if(ch!='.'){
                            if(count[ch]!=0){
                                return false;
                            }
                            count[ch]=1;
                        }
                    }
                }
            }    
        }
        return true;
        
    }
};

File: 0036-valid-sudoku/0036-valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        #Brute Force
        #Checking for column
        for row in range(9):
            check=[0]*10
            for col in range(9):
                if board[row][col] !=".":
                    ele=ord(board[row][col])-ord("0")
                    if check[ele]:
                        print(row,col)
                        return False
                    check[ele]=1
        #Checking for row
        for col in range(9):
            check=[0]*10
            for row in range(9):
                if board[row][col]!=".":
                    ele=ord(board[row][col])-ord("0")
                    if check[ele]:
                        return False
                    check[ele]=1
        #Checking for 3*3 box
        for i in range(3):
            for j in range(3):
                check=[0]*10
                for row in range(3):
                    for col in range(3):
                        if board[i*3+row][j*3+col]!=".":
                            ele=ord(board[i*3+row][j*3+col])-ord("0")
                            if check[ele]:
                                return False
                            check[ele]=1
        return True